Chapter07/Exercise7.02/running_average.clj

```diff
@@ -1,19 +1,36 @@
 ;;; In REPL: 2. Simulate a source of potatoes
-(def endless-potatoes (repeatedly (fn [] (+ 10 (rand-int 390)))))
+(def endless-potatoes
+  (repeatedly (fn [] (+ 10 (rand-int 390)))))
 
 ;;; In REPL: 3. Test
 (take 5 endless-potatoes)
+;; => (76 90 67 224 375)
+
 (take 10 endless-potatoes)
+;; => (76 90 67 224 375 273 278 288 374 70)
+
+[200 5 784]
+
+'([59 1 59] [134 2 193] [358 3 551])
 
+(double (/ 551 3))
+;; => 183.6666666666667
 
 ;;; In REPL: 7. The complete function
-(defn average-potatoes [so-far arrivals]
+(defn average-potatoes [prev arrivals]
```

Comment:
Changing the name of the `so-far` argument here will create a difference with the book that might be confusing for readers. I would recommend leaving `so-far`. The `so-far` name was used in Chapter 5 as well, so there might be some familiarity.

Reply:
Hi @josf here `so-far` referred to all the potatoes that were collected `so-far`, and then `first` was called to get the latest. In this case `first` is called before the recursive call, so it does not refer to `so-far` anymore, but only the `prev` element. 

Also `so-far` was returned when the `arrivals` were empty, and I don't think that does what it is supposed to do since it will `cons` op all potatoes `so-far` again when `arrivals` is empty.

Note that I followed the text in the book for exercise 7.02 that does use `prev` as well.

Because the code now works and it follows the code in the video I would like to keep it as is. 

Is that okay with you?
Regarding the above, is that okay to leave this unchanged?

Reply:
I see the version in GitHub has the elements in reverse order (so highest running average first) and the new version does not. 

In the changed function `first` is called on the `arrivals` before the recursive call so then the `prev` name is correct. Note that I did use this code in the screenrecording. I followed the code in the draft for this exercise and there it was using `prev`. 

Let me know regarding the above information if you still want to change it back then I'll do that.

Reply:
I tried adding both versions to the solution and then I noticed something. The old version doesn't terminate when you pass it the `endless-potatoes` as an argument. Makes sense because when you reverse an infinite stream you have to wait an infinite amount of time. Maybe the code is changed because it was fixed?

Reply:
I think I understand better now. The book might be a little bit ahead of this version.

Please go ahead and merge your changes. I'll check as soon as I get a chance so that we can update the book if necessary. I think that might still be possible for perhaps a few more hours...

Thanks again for going through the code so thoroughly!

Reply:
Hehe my pleasure I learned a lot about lazy sequences ;). I do not have merge rights. Can you merge? Good luck with the writing!

Reply:
I finally had the time to go over this in detail. Yes, the version here was old and broken, and the book version had already been corrected. I'm keeping your changes and adding the book version next to them, since the book version uses `next` and not `rest`, and there's some discussion of that difference.

So all is well that ends well, I think.